File: server/app/api/public/ziliao.py
from app.api.public._deps import *  # noqa: F401,F403

router = APIRouter()
# ===== 资料分析 =====


@router.get("/ziliao/overview")
def ziliao_overview(
    user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    return ApiResponse.ok(get_ziliao_overview(db, user.id).model_dump())


@router.get("/ziliao/formulas")
def ziliao_formulas(
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    return ApiResponse.ok([x.model_dump() for x in list_ziliao_formulas(db)])


@router.get("/ziliao/formulas/{formula_id}")
def ziliao_formula_detail(
    formula_id: str,
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    out = get_ziliao_formula(db, formula_id)
    if not out:
        return ApiResponse.fail("公式不存在", code=404)
    return ApiResponse.ok(out.model_dump())


@router.get("/ziliao/types")
def ziliao_types(
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    return ApiResponse.ok([x.model_dump() for x in list_ziliao_types(db)])


@router.get("/ziliao/types/{type_id}")
def ziliao_type_detail(
    type_id: str,
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    out = get_ziliao_type(db, type_id)
    if not out:
        return ApiResponse.fail("题型不存在", code=404)
    return ApiResponse.ok(out.model_dump())


@router.get("/ziliao/tricks")
def ziliao_tricks(
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    return ApiResponse.ok([x.model_dump() for x in list_ziliao_tricks(db)])


@router.get("/ziliao/tricks/{trick_id}")
def ziliao_trick_detail(
    trick_id: str,
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    out = get_ziliao_trick(db, trick_id)
    if not out:
        return ApiResponse.fail("技巧不存在", code=404)
    return ApiResponse.ok(out.model_dump())


@router.get("/ziliao/drill/sets")
def ziliao_drill_sets(
    typeCode: str | None = None,
    includeSample: bool | None = None,
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    return ApiResponse.ok(
        [
            x.model_dump()
            for x in list_ziliao_drill_sets(db, type_code=typeCode, include_sample=includeSample)
        ]
    )


@router.get("/ziliao/drill/set/{set_id}")
def ziliao_drill_set_detail(
    set_id: str,
    _user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    out = get_ziliao_drill_set(db, set_id)
    if not out:
        return ApiResponse.fail("练习组不存在", code=404)
    return ApiResponse.ok(out.model_dump())


@router.post("/ziliao/drill/submit")
def ziliao_drill_submit(
    body: ZiliaoDrillSubmitIn,
    user: AppUser = Depends(get_app_user),
    db: Session = Depends(get_db),
):
    out = submit_ziliao_drill(db, user, body)
    if not out:
        return ApiResponse.fail("练习组不存在", code=404)
    return ApiResponse.ok(out.model_dump())

File: src/api/domains/ziliao.ts
import * as d from '../_shared'

export const apiZiliao = {
  // ===== 资料分析 =====

  getZiliaoOverview(): Promise<d.ApiRes<d.ZiliaoOverview>> {
    return d.isMock ? d.mockService.getZiliaoOverview() : d.request('/api/ziliao/overview')
  },

  listZiliaoFormulas(): Promise<d.ApiRes<d.ZiliaoFormula[]>> {
    return d.isMock ? d.mockService.listZiliaoFormulas() : d.request('/api/ziliao/formulas')
  },

  getZiliaoFormula(id: string): Promise<d.ApiRes<d.ZiliaoFormula>> {
    return d.isMock ? d.mockService.getZiliaoFormula(id) : d.request(`/api/ziliao/formulas/${id}`)
  },

  listZiliaoTypes(): Promise<d.ApiRes<d.ZiliaoQuestionType[]>> {
    return d.isMock ? d.mockService.listZiliaoTypes() : d.request('/api/ziliao/types')
  },

  getZiliaoType(id: string): Promise<d.ApiRes<d.ZiliaoQuestionType>> {
    return d.isMock ? d.mockService.getZiliaoType(id) : d.request(`/api/ziliao/types/${id}`)
  },

  listZiliaoTricks(): Promise<d.ApiRes<d.ZiliaoTrick[]>> {
    return d.isMock ? d.mockService.listZiliaoTricks() : d.request('/api/ziliao/tricks')
  },

  getZiliaoTrick(id: string): Promise<d.ApiRes<d.ZiliaoTrick>> {
    return d.isMock ? d.mockService.getZiliaoTrick(id) : d.request(`/api/ziliao/tricks/${id}`)
  },

  listZiliaoDrillSets(
    typeCode?: string,
    includeSample?: boolean,
  ): Promise<d.ApiRes<d.ZiliaoDrillSet[]>> {
    if (d.isMock) return d.mockService.listZiliaoDrillSets(typeCode)
    const qs: string[] = []
    if (typeCode) qs.push(`typeCode=${encodeURIComponent(typeCode)}`)
    if (includeSample !== undefined) qs.push(`includeSample=${includeSample ? 'true' : 'false'}`)
    const q = qs.length ? `?${qs.join('&')}` : ''
    return d.request(`/api/ziliao/drill/sets${q}`)
  },

  getZiliaoDrillSet(setId: string): Promise<d.ApiRes<d.ZiliaoDrillSetDetail>> {
    return d.isMock
      ? d.mockService.getZiliaoDrillSet(setId)
      : d.request(`/api/ziliao/drill/set/${encodeURIComponent(setId)}`)
  },

  submitZiliaoDrill(data: {
    setId: string
    answers: { questionId: string; userAnswer: string | string[] }[]
    timeUsedSec?: number
    typeCode?: string
    saveWrongs?: boolean
  }): Promise<d.ApiRes<d.ZiliaoDrillSubmitResult>> {
    return d.isMock
      ? d.mockService.submitZiliaoDrill(data)
      : d.request('/api/ziliao/drill/submit', { method: 'POST', data })
  },

  getCountdown(): Promise<d.ApiRes<d.ExamCountdown | null>> {
    return d.isMock ? d.mockService.getCountdown() : d.request('/api/countdown')
  },

  saveCountdown(data: {
    examName: string
    examDate: string
    note?: string
  }): Promise<d.ApiRes<d.ExamCountdown>> {
    return d.isMock
      ? d.mockService.saveCountdown(data)
      : d.request('/api/countdown', { method: 'PUT', data })
  },

  deleteCountdown(): Promise<d.ApiRes<{ deleted: boolean }>> {
    return d.isMock
      ? d.mockService.deleteCountdown()
      : d.request('/api/countdown', { method: 'DELETE' })
  },
}
